book/content/material/material_basics.md

```diff
@@ -0,0 +1,198 @@
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+import festim as F
+from dolfinx.mesh import create_unit_square
+from mpi4py import MPI
+
+fenics_mesh = create_unit_square(MPI.COMM_WORLD, 10, 10)
+festim_mesh = F.Mesh(fenics_mesh)
+```
+
+To define one material on each subdomain:
+
+```{code-cell} ipython3
+mat1 = F.Material(D_0=1.11e-6, E_D=0.4)  # m2/s, eV
+
+top = F.VolumeSubdomain(id=1, material=mat, locator=lambda x: x[0] >= 0.5)
+bottom = F.VolumeSubdomain(id=2, material=mat, locator=lambda x: x[0] < 0.5)
+```
+
+Similarly, for two materials:
+
+```{code-cell} ipython3
+mat1 = F.Material(D_0=1.11e-6, E_D=0.4)  # m2/s, eV
+mat2 = F.Material(D_0=2e-6, E_D=0.3)  # m2/s, eV
+
+top = F.VolumeSubdomain(id=1, material=mat1, locator=lambda x: x[0] >= 0.5)
+bottom = F.VolumeSubdomain(id=2, material=mat2, locator=lambda x: x[0] < 0.5)
+```
+
+## Solving a multi-material diffusion problem ##
+Considering the following 2D example, where hydrogen diffuses through a 2D domain composed of two materials with different diffusion and solubility properties. The top half (Material A) has a higher diffusion coefficient and solubility than the bottom half (Material B). The interface at $𝑦=0.5$ clearly separates the two materials, and the steady-state hydrogen distribution illustrates how material properties impact transport.
+
++++
+
+First, create the mesh:
+
+```{code-cell} ipython3
+import festim as F
+import numpy as np
+from mpi4py import MPI
+from dolfinx.mesh import create_unit_square
+import pyvista
+from dolfinx import plot
+
+fenics_mesh = create_unit_square(MPI.COMM_WORLD, 40, 40)  # finer resolution
+festim_mesh = F.Mesh(fenics_mesh)
+```
+
+Then, we define materials:
+
+```{code-cell} ipython3
+# Top material (Material A) - "Cu-like"
+material_A = F.Material(
+    name="Material_A",
+    D_0=1e-6,    # m²/s
+    E_D=0.3,     # eV
+    K_S_0=5e-3,    # mol/m³/Pa  (solubility prefactor)
+    E_K_S=0.1,     # eV (activation energy for solubility)
+    thermal_conductivity=400,  # W/m/K
+    heat_capacity=385, # J/kg/K
+    density=8960       # kg/m³
+)
+
+# Bottom material (Material B) - "W-like"
+material_B = F.Material(
+    name="Material_B",
+    D_0=2e-7,    # m²/s
+    E_D=1.2,     # eV
+    K_S_0=2e-3,    # mol/m³/Pa
+    E_K_S=0.2,     # eV
+    thermal_conductivity=180,  # W/m/K
+    heat_capacity=140, # J/kg/K
+    density=19300      # kg/m³
+)
+```
+
+Now subdomains:
+
+```{code-cell} ipython3
+top_volume = F.VolumeSubdomain(id=1, material=material_A, locator=lambda x: x[1] >= 0.5)
+bottom_volume = F.VolumeSubdomain(id=2, material=material_B, locator=lambda x: x[1] <= 0.5)
+
+top_surface = F.SurfaceSubdomain(id=1, locator=lambda x: np.isclose(x[1], 1.0))
+bottom_surface = F.SurfaceSubdomain(id=2, locator=lambda x: np.isclose(x[1], 0.0))
+```
+
+Finally, we define the problem, boundary conditions, and then solve:
+
+```{code-cell} ipython3
+
+my_model = F.HydrogenTransportProblem()
```

Comment:
If you are solving a multi-material case you should use `HydrogenTransportProblemDiscontinuous` otherwise you won't see a jump in concentration at the interface